src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -2745,6 +2746,200 @@ namespace QgsWms
+                           << QStringLiteral( "AND" ) << QgsExpression::quotedColumnRef( dim.endFieldName )
+                           << QStringLiteral( "<" ) << QgsExpression::quotedValue( rangeMin )
+                           << QStringLiteral( ") AND NOT (" ) << QgsExpression::quotedColumnRef( dim.fieldName )
+                           << QStringLiteral( ">" ) << QgsExpression::quotedValue( rangeMax )
+                           << QStringLiteral( "AND" ) << QgsExpression::quotedColumnRef( dim.endFieldName )
+                           << QStringLiteral( ">" ) << QgsExpression::quotedValue( rangeMax )
+                           << QStringLiteral( ")" );
+                }
+                dimExplist << expElems.join( ' ' );
+              }
+              else
+              {
+                QVariant dimVariant = QVariant( dimValue );
+                if ( !dimField.convertCompatible( dimVariant ) )
+                {
+                  continue; // throw an error
+                }
+                // Build expression for this value
+                if ( endFieldIndex == -1 )
+                {
+                  // Field is equal to
+                  dimExplist << QgsExpression::createFieldEqualityExpression( dim.fieldName, dimVariant );
+                }
+                else
+                {
+                  // The start field is lesser or equal to
+                  // and the end field is greater or equal to
+                  QStringList expElems;
+                  expElems << QgsExpression::quotedColumnRef( dim.fieldName )
+                           << QStringLiteral( "<=" ) << QgsExpression::quotedValue( dim.referenceValue )
+                           << QStringLiteral( "AND" ) << QgsExpression::quotedColumnRef( dim.endFieldName )
+                           << QStringLiteral( ">=" ) << QgsExpression::quotedValue( dim.referenceValue );
+                  dimExplist << expElems.join( ' ' );
+                }
+              }
+            }
+            // Build the expression for this dimension
+            if ( expList.size() == 1 )
+            {
+              expList << dimExplist;
+            }
+            else
+            {
+              expList << QStringLiteral( "( %s )" ).arg( dimExplist.join( QStringLiteral( " ) AND ( " ) ) );
+            }
+          }
+        }
+      }
+
+      // Join and apply expressions provided by OGC filter and Dimensions
+      QString exp;
+      if ( expList.size() == 1 )
+      {
+        exp = expList[0];
+      }
+      else if ( expList.size() > 1 )
+      {
+        exp = QStringLiteral( "( %s )" ).arg( expList.join( QStringLiteral( " ) AND ( " ) ) );
+      }
+      if ( !exp.isEmpty() )
+      {
+        std::unique_ptr<QgsExpression> expression( new QgsExpression( exp ) );
+        if ( expression )
+        {
+          mFeatureFilter.setFilter( filteredLayer, *expression );
+        }
+      }
```

Comment:
I think that it would be nice to have a dedicated method to process dimensions filtering. What do you think?

Reply:
The `QgsFeatureFilter` has not been design to update expression for filteredLayer and Dimension is a way to filter a layer. Dimension is applied excatly for the same request as Filter.

Reply:
Actually, it's just that the `QgsRenderer::setLayerFilter` method is starting to be a little too long for being easily read.

What do you think about defining a new method `QgsRenderer::dimensionFilter` which returns a `QStringList` for a specific layer?

Reply:
Done